File: lib/services/city_service.js
import { db } from "./firestore_connect";
import { collection, getDocs } from "firebase/firestore";
import { getDatabase, ref, get } from "firebase/database";

// Firestore
export async function fetchCitiesFirestore() {
  try {
    const snapshot = await getDocs(collection(db, "cities"));
    return snapshot.docs.map(doc => ({ id: doc.id, ...doc.data() }));
  } catch (e) {
    console.error("Error fetching cities from Firestore:", e);
    return [];
  }
}

// Realtime Database
// export async function fetchCitiesRTDB() {
//   try {
//     const rtdb = getDatabase();
//     const snapshot = await get(ref(rtdb, "cities"));
//     if (snapshot.exists()) {
//       const data = snapshot.val();
//       return Object.entries(data).map(([id, value]) => ({ id, ...value }));
//     }
//     return [];
//   } catch (e) {
//     console.error("Error fetching cities from RTDB:", e);
//     return [];
//   }
// }

File: lib/services/city_service.dart
import 'firestore_connect.dart';

class CityService {
  static Future<List<Map<String, dynamic>>> fetchCitiesFirestore() async {
    final snap = await FirestoreConnect.db.collection('cities').get();
    return snap.docs.map((d) {
      final data = d.data();
      data['id'] = d.id;
      return data;
    }).toList();
  }

  static Stream<List<Map<String, dynamic>>> streamCities(String cityId) {
    return FirestoreConnect.db
        .collection('cities')
        .snapshots()
        .map((snap) => snap.docs.map((d) => d.data()).toList());
  }

  static Future<Map<String, dynamic>?> fetchCityById(String id) async {
    final doc = await FirestoreConnect.db.collection('cities').doc(id).get();
    return doc.data();
  }

  static Stream<Map<String, dynamic>?> streamCityById(String id) {
    return FirestoreConnect.db
        .collection('cities')
        .doc(id)
        .snapshots()
        .map((d) => d.data());
  }
}
